// @pulumi/openstack/config/vars.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
const __config = new pulumi.Config("openstack");
Object.defineProperty(exports, "allowReauth", {
    get() {
        var _a;
        return (_a = __config.getObject("allowReauth")) !== null && _a !== void 0 ? _a : utilities.getEnvBoolean("OS_ALLOW_REAUTH");
    },
    enumerable: true,
});
Object.defineProperty(exports, "applicationCredentialId", {
    get() {
        return __config.get("applicationCredentialId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "applicationCredentialName", {
    get() {
        return __config.get("applicationCredentialName");
    },
    enumerable: true,
});
Object.defineProperty(exports, "applicationCredentialSecret", {
    get() {
        return __config.get("applicationCredentialSecret");
    },
    enumerable: true,
});
Object.defineProperty(exports, "authUrl", {
    get() {
        return __config.get("authUrl");
    },
    enumerable: true,
});
Object.defineProperty(exports, "cacertFile", {
    get() {
        return __config.get("cacertFile");
    },
    enumerable: true,
});
Object.defineProperty(exports, "cert", {
    get() {
        return __config.get("cert");
    },
    enumerable: true,
});
Object.defineProperty(exports, "cloud", {
    get() {
        var _a;
        return (_a = __config.get("cloud")) !== null && _a !== void 0 ? _a : utilities.getEnv("OS_CLOUD");
    },
    enumerable: true,
});
Object.defineProperty(exports, "defaultDomain", {
    get() {
        return __config.get("defaultDomain");
    },
    enumerable: true,
});
Object.defineProperty(exports, "delayedAuth", {
    get() {
        var _a;
        return (_a = __config.getObject("delayedAuth")) !== null && _a !== void 0 ? _a : utilities.getEnvBoolean("OS_DELAYED_AUTH");
    },
    enumerable: true,
});
Object.defineProperty(exports, "disableNoCacheHeader", {
    get() {
        return __config.getObject("disableNoCacheHeader");
    },
    enumerable: true,
});
Object.defineProperty(exports, "domainId", {
    get() {
        return __config.get("domainId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "domainName", {
    get() {
        return __config.get("domainName");
    },
    enumerable: true,
});
Object.defineProperty(exports, "enableLogging", {
    get() {
        return __config.getObject("enableLogging");
    },
    enumerable: true,
});
Object.defineProperty(exports, "endpointOverrides", {
    get() {
        return __config.getObject("endpointOverrides");
    },
    enumerable: true,
});
Object.defineProperty(exports, "endpointType", {
    get() {
        var _a;
        return (_a = __config.get("endpointType")) !== null && _a !== void 0 ? _a : utilities.getEnv("OS_ENDPOINT_TYPE");
    },
    enumerable: true,
});
Object.defineProperty(exports, "insecure", {
    get() {
        var _a;
        return (_a = __config.getObject("insecure")) !== null && _a !== void 0 ? _a : utilities.getEnvBoolean("OS_INSECURE");
    },
    enumerable: true,
});
Object.defineProperty(exports, "key", {
    get() {
        return __config.get("key");
    },
    enumerable: true,
});
Object.defineProperty(exports, "maxRetries", {
    get() {
        return __config.getObject("maxRetries");
    },
    enumerable: true,
});
Object.defineProperty(exports, "password", {
    get() {
        return __config.get("password");
    },
    enumerable: true,
});
Object.defineProperty(exports, "projectDomainId", {
    get() {
        return __config.get("projectDomainId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "projectDomainName", {
    get() {
        return __config.get("projectDomainName");
    },
    enumerable: true,
});
Object.defineProperty(exports, "region", {
    get() {
        var _a;
        return (_a = __config.get("region")) !== null && _a !== void 0 ? _a : utilities.getEnv("OS_REGION_NAME");
    },
    enumerable: true,
});
Object.defineProperty(exports, "swauth", {
    get() {
        var _a;
        return (_a = __config.getObject("swauth")) !== null && _a !== void 0 ? _a : utilities.getEnvBoolean("OS_SWAUTH");
    },
    enumerable: true,
});
Object.defineProperty(exports, "tenantId", {
    get() {
        return __config.get("tenantId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "tenantName", {
    get() {
        return __config.get("tenantName");
    },
    enumerable: true,
});
Object.defineProperty(exports, "token", {
    get() {
        return __config.get("token");
    },
    enumerable: true,
});
Object.defineProperty(exports, "useOctavia", {
    get() {
        var _a;
        return (_a = __config.getObject("useOctavia")) !== null && _a !== void 0 ? _a : utilities.getEnvBoolean("OS_USE_OCTAVIA");
    },
    enumerable: true,
});
Object.defineProperty(exports, "userDomainId", {
    get() {
        return __config.get("userDomainId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "userDomainName", {
    get() {
        return __config.get("userDomainName");
    },
    enumerable: true,
});
Object.defineProperty(exports, "userId", {
    get() {
        return __config.get("userId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "userName", {
    get() {
        return __config.get("userName");
    },
    enumerable: true,
});
